<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Data Processing, Checking, and Exploration</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Fiber_Intervention_Study</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data Processing, Checking, and Exploration</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-02-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Fiber_Intervention_Study/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191210code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191210)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191210code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191210)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabFiberInterventionStudytreea6488a7874658c4e5441be28019b77458d55583btargetblanka6488a7a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/GreathouseLab/Fiber_Intervention_Study/tree/a6488a7874658c4e5441be28019b77458d55583b" target="_blank">a6488a7</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabFiberInterventionStudytreea6488a7874658c4e5441be28019b77458d55583btargetblanka6488a7a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/Microbiome Data/NoahFolder/ASA24_fiber_F_V_intake_by_week.xlsx
    Untracked:  data/Microbiome Data/NoahFolder/FFQ_HEI_Fiber.xlsx
    Untracked:  fig/predicted_Actinobacteria.png
    Untracked:  fig/predicted_Bacteroidetes.png
    Untracked:  fig/predicted_Cyanobacteria.png
    Untracked:  fig/predicted_Euryarchaeota.png
    Untracked:  fig/predicted_Firmicutes.png
    Untracked:  fig/predicted_Fusobacteria.png
    Untracked:  fig/predicted_Lentisphaerae.png
    Untracked:  fig/predicted_Proteobacteria.png
    Untracked:  fig/predicted_Tenericutes.png
    Untracked:  fig/predicted_Verrucomicrobia.png

Unstaged changes:
    Modified:   analysis/alpha_diversity.Rmd
    Modified:   analysis/data_processing.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/lme_alpha.Rmd
    Modified:   analysis/lme_microbiome.Rmd
    Modified:   code/get_cleaned_data.R
    Deleted:    predicted_Actinobacteria.png
    Deleted:    predicted_Bacteroidetes.png
    Deleted:    predicted_Cyanobacteria.png
    Deleted:    predicted_Euryarchaeota.png
    Deleted:    predicted_Firmicutes.png
    Deleted:    predicted_Fusobacteria.png
    Deleted:    predicted_Lentisphaerae.png
    Deleted:    predicted_Proteobacteria.png
    Deleted:    predicted_Tenericutes.png
    Deleted:    predicted_Verrucomicrobia.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p><em>Make table of sequencial cut-offs where we removed OTUs</em></p>
<p>This page contains the investigation of the raw data (OTUs) to identify if outliers are present or whether other issues emerge that may influence our results in unexpected ways. This file goes through the following checks:</p>
<ol style="list-style-type: decimal">
<li>Removal of Phylum NA features</li>
<li>Computation of total and average prevalence in each Phylum</li>
<li>Removal Phyla with 1% or less of all samples</li>
<li>Computation of total read count for each Phyla</li>
<li>Plotting taxa prevalence vs total counts - identify a natural threshold if clear, if not use 5%</li>
<li>Merging taxa to genus rank/level</li>
<li>Abundance Value Transformations</li>
<li>Plotting of abundance values by “Intervention A or B” before transformation and after</li>
<li>Checking of any bimodal distributions using “subset_taxa” function and plot by “intervention”</li>
</ol>
<div id="taxonomic-filtering" class="section level1">
<h1>Taxonomic Filtering</h1>
<div id="removal-of-phylum-na-features" class="section level2">
<h2>1. Removal of Phylum NA features</h2>
<pre class="r"><code># show ranks
rank_names(phylo_data0)</code></pre>
<pre><code>[1] &quot;Kingdom&quot; &quot;Phylum&quot;  &quot;Class&quot;   &quot;Order&quot;   &quot;Family&quot;  &quot;Genus&quot;  </code></pre>
<pre class="r"><code># table of features for each phylum
table(tax_table(phylo_data0)[,&quot;Phylum&quot;], exclude=NULL)</code></pre>
<pre><code>
    __Actinobacteria      __Bacteroidetes      __Cyanobacteria 
                  19                   49                    5 
__Epsilonbacteraeota      __Euryarchaeota         __Firmicutes 
                   1                    2                  334 
      __Fusobacteria      __Lentisphaerae     __Proteobacteria 
                   2                    3                   23 
     __Synergistetes        __Tenericutes    __Verrucomicrobia 
                   1                   12                    1 </code></pre>
<p>Note that no taxa were labels as <em>NA</em> so none were removed.</p>
</div>
<div id="computation-of-total-and-average-prevalence-in-each-phylum" class="section level2">
<h2>2. Computation of total and average prevalence in each Phylum</h2>
<pre class="r"><code># compute prevalence of each feature
prevdf &lt;- apply(X=otu_table(phylo_data0), 
                MARGIN= ifelse(taxa_are_rows(phylo_data0), yes=1, no=2),
                FUN=function(x){sum(x&gt;0)})
# store as data.frame with labels
prevdf &lt;- data.frame(Prevalence=prevdf,
                     TotalAbundance=taxa_sums(phylo_data0),
                     tax_table(phylo_data0))</code></pre>
<p>Now we get to compute the totals and averages.</p>
<pre class="r"><code>totals &lt;- plyr::ddply(prevdf, &quot;Phylum&quot;,
            function(df1){
              A &lt;- cbind(mean(df1$Prevalence), sum(df1$Prevalence))
              colnames(A) &lt;- c(&quot;Average&quot;, &quot;Total&quot;)
              A
              }
            ) # end

totals</code></pre>
<pre><code>                 Phylum   Average Total
1      __Actinobacteria  6.842105   130
2       __Bacteroidetes 10.816327   530
3       __Cyanobacteria  4.000000    20
4  __Epsilonbacteraeota  5.000000     5
5       __Euryarchaeota  8.000000    16
6          __Firmicutes 11.395210  3806
7        __Fusobacteria  4.000000     8
8       __Lentisphaerae 11.333333    34
9      __Proteobacteria 10.826087   249
10      __Synergistetes  5.000000     5
11        __Tenericutes  4.750000    57
12    __Verrucomicrobia 29.000000    29</code></pre>
<p>The Phylum that appear to be quite low in abundance are Cyanobacteria, Epsilonbacteraeota, Euryarchaeota, Fusobacteria and Synergistetes. However, any of the taxa under a total of 100 may be suspect. First, we will remove the taxa that are clearly too low in abudance (&lt;=5).</p>
<pre class="r"><code>filterPhyla &lt;- totals$Phylum[totals$Total &lt;= 5, drop=T] # drop allows some of the attributes to be removed

phylo_data1 &lt;- subset_taxa(phylo_data0, !Phylum %in% filterPhyla)
phylo_data1</code></pre>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 450 taxa and 37 samples ]
sample_data() Sample Data:       [ 37 samples by 90 sample variables ]
tax_table()   Taxonomy Table:    [ 450 taxa by 6 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 450 tips and 449 internal nodes ]</code></pre>
<p>Next, we explore the taxa in more detail next as we move to remove some of these low abundance taxa.</p>
</div>
<div id="removal-phyla-with-1-or-less-of-all-samples-prevalence-filtering" class="section level2">
<h2>3. Removal Phyla with 1% or less of all samples (prevalence filtering)</h2>
<pre class="r"><code>prevdf1 &lt;- subset(prevdf, Phylum %in% get_taxa_unique(phylo_data1, &quot;Phylum&quot;))
ggplot(prevdf1, aes(TotalAbundance+1,
                    Prevalence/nsamples(phylo_data0))) +
  geom_hline(yintercept=0.01, alpha=0.5, linetype=2)+
  geom_point(size=2, alpha=0.75) +
  scale_x_log10()+
  labs(x=&quot;Total Abundance&quot;, y=&quot;Prevalance [Frac. Samples]&quot;)+
  facet_wrap(.~Phylum) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/data_processing.Rmd/3-remove-phylum-1-1.png" width="225%" style="display: block; margin: auto;" /></p>
<p>Note: for plotting purposes, a <span class="math inline">\(+1\)</span> was added to all TotalAbundances to avoid a taking the log of 0.</p>
<p>Next, we define a prevalence threshold, that way the taxa can be pruned to a prespecified level. In this study, we used 0.05 (5%) of total samples.</p>
<pre class="r"><code>prevalenceThreshold &lt;- 0.01*nsamples(phylo_data0)
prevalenceThreshold</code></pre>
<pre><code>[1] 0.37</code></pre>
<pre class="r"><code># execute the filtering to this level
keepTaxa &lt;- rownames(prevdf1)[(prevdf1$Prevalence &gt;= prevalenceThreshold)]
phylo_data2 &lt;- prune_taxa(keepTaxa, phylo_data1)</code></pre>
</div>
<div id="merge-taxa-to-genus-level" class="section level2">
<h2>4. Merge taxa (to genus level)</h2>
<pre class="r"><code>genusNames &lt;- get_taxa_unique(phylo_data2, &quot;Genus&quot;)
#phylo_data3 &lt;- merge_taxa(phylo_data2, genusNames, genusNames[which.max(taxa_sums(phylo_data2)[genusNames])])


# How many genera would be present after filtering?
length(get_taxa_unique(phylo_data2, taxonomic.rank = &quot;Genus&quot;))</code></pre>
<pre><code>[1] 159</code></pre>
<pre class="r"><code>## [1] 49

phylo_data3 = tax_glom(phylo_data2, &quot;Genus&quot;, NArm = TRUE)</code></pre>
</div>
<div id="relative-adbundance-plot" class="section level2">
<h2>4. Relative Adbundance Plot</h2>
<pre class="r"><code>plot_abundance = function(physeq, title = &quot;&quot;, ylab=&quot;Abundance&quot;){
  # Arbitrary subset, based on Phylum, for plotting
  #p1f = subset_taxa(physeq, Phylum %in% &quot;__Firmicutes&quot;)
  mphyseq = psmelt(physeq)
  mphyseq &lt;- subset(mphyseq, Abundance &gt; 0)
  ggplot(data = mphyseq, mapping = aes_string(x = &quot;Intervention&quot;, y = &quot;Abundance&quot;)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position=&quot;none&quot;)
}


# Transform to relative abundance. Save as new object.
phylo_data3ra = transform_sample_counts(phylo_data3, function(x){x / sum(x)})


plotBefore = plot_abundance(phylo_data3, ylab=&quot;Abundance prior to transformation&quot;)
plotAfter = plot_abundance(phylo_data3ra, ylab=&quot;Relative Abundance&quot;)
# Combine each plot into one graphic.
grid.arrange(nrow = 2, plotBefore, plotAfter)</code></pre>
<p><img src="figure/data_processing.Rmd/4-rel-abund-1.png" width="225%" style="display: block; margin: auto;" /></p>
</div>
<div id="abundance-by-phylum" class="section level2">
<h2>Abundance by Phylum</h2>
<pre class="r"><code>plot_abundance = function(physeq, title = &quot;&quot;, Facet = &quot;Phylum&quot;, ylab=&quot;Abundance&quot;){
  # Arbitrary subset, based on Phylum, for plotting
  #p1f = subset_taxa(physeq, Phylum %in% &quot;__Firmicutes&quot;)
  mphyseq = psmelt(physeq)
  mphyseq &lt;- subset(mphyseq, Abundance &gt; 0)
  ggplot(data = mphyseq, mapping = aes_string(x = &quot;Intervention&quot;, y = &quot;Abundance&quot;)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position=&quot;none&quot;)
}



plotBefore = plot_abundance(phylo_data3, ylab=&quot;Abundance prior to transformation&quot;)
plotAfter = plot_abundance(phylo_data3ra, ylab=&quot;Relative Abundance&quot;)
# Combine each plot into one graphic.
grid.arrange(nrow = 2, plotBefore, plotAfter)</code></pre>
<p><img src="figure/data_processing.Rmd/4-phylum-1.png" width="225%" style="display: block; margin: auto;" /> Now, let’s dive into the abundances in more detail. We will investigate the <em>bacteroidetes, firmicute, verrucomicrobia and proteobacteria</em> in more detail (down to the Order).</p>
<div id="phylum-bacteroidetes" class="section level3">
<h3>Phylum: Bacteroidetes</h3>
<pre class="r"><code>plot_abundance = function(physeq, title = &quot;&quot;, Facet = &quot;Order&quot;, ylab=&quot;Abundance&quot;){
  # Arbitrary subset, based on Phylum, for plotting
  p1f = subset_taxa(physeq, Phylum %in%  &quot;__Bacteroidetes&quot;)
  mphyseq = psmelt(p1f)
  mphyseq &lt;- subset(mphyseq, Abundance &gt; 0)
  ggplot(data = mphyseq, mapping = aes_string(x = &quot;Intervention&quot;, y = &quot;Abundance&quot;)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position=&quot;none&quot;)
}


plotBefore = plot_abundance(phylo_data3,
                            ylab=&quot;Abundance prior to transformation&quot;)
plotAfter = plot_abundance(phylo_data3ra,
                           ylab=&quot;Relative Abundance&quot;)
# Combine each plot into one graphic.
grid.arrange(nrow = 2, plotBefore, plotAfter)</code></pre>
<p><img src="figure/data_processing.Rmd/4-rel-abund-2-1.png" width="225%" style="display: block; margin: auto;" /></p>
<p>Flav. was only present in intervention group A.</p>
</div>
<div id="phylum-firmicutes" class="section level3">
<h3>Phylum: Firmicutes</h3>
<pre class="r"><code>plot_abundance = function(physeq, title = &quot;&quot;, Facet = &quot;Order&quot;, ylab=&quot;Abundance&quot;){
  # Arbitrary subset, based on Phylum, for plotting
  p1f = subset_taxa(physeq, Phylum %in%  &quot;__Firmicutes&quot;)
  mphyseq = psmelt(p1f)
  mphyseq &lt;- subset(mphyseq, Abundance &gt; 0)
  ggplot(data = mphyseq, mapping = aes_string(x = &quot;Intervention&quot;, y = &quot;Abundance&quot;)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position=&quot;none&quot;)
}


plotBefore = plot_abundance(phylo_data3,
                            ylab=&quot;Abundance prior to transformation&quot;)
plotAfter = plot_abundance(phylo_data3ra,
                           ylab=&quot;Relative Abundance&quot;)
# Combine each plot into one graphic.
grid.arrange(nrow = 2, plotBefore, plotAfter)</code></pre>
<p><img src="figure/data_processing.Rmd/4-rel-abund-3-1.png" width="225%" style="display: block; margin: auto;" /></p>
<div id="order-selenomonadales" class="section level4">
<h4>Order: Selenomonadales</h4>
<pre class="r"><code>plot_abundance = function(physeq, title = &quot;&quot;, Facet = &quot;Genus&quot;, ylab=&quot;Abundance&quot;){
  # Arbitrary subset, based on Phylum, for plotting
  p1f = subset_taxa(physeq, Phylum %in%  &quot;__Firmicutes&quot; &amp; Order %in% &quot;__Selenomonadales&quot; &amp; Family %in% &quot;__Veillonellaceae&quot;)
  mphyseq = psmelt(p1f)
  mphyseq &lt;- subset(mphyseq, Abundance &gt; 0)
  ggplot(data = mphyseq, mapping = aes_string(x = &quot;Intervention&quot;, y = &quot;Abundance&quot;)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position=&quot;none&quot;)
}


plotBefore = plot_abundance(phylo_data3,
                            ylab=&quot;Abundance prior to transformation&quot;)
plotAfter = plot_abundance(phylo_data3ra,
                           ylab=&quot;Relative Abundance&quot;)
# Combine each plot into one graphic.
grid.arrange(nrow = 2, plotBefore, plotAfter)</code></pre>
<p><img src="figure/data_processing.Rmd/unnamed-chunk-1-1.png" width="225%" style="display: block; margin: auto;" /></p>
<div id="family-veillonellaceae" class="section level5">
<h5>Family: Veillonellaceae</h5>
<pre class="r"><code>plot_abundance = function(physeq, title = &quot;&quot;, Facet = &quot;Genus&quot;, ylab=&quot;Abundance&quot;){
  # Arbitrary subset, based on Phylum, for plotting
  p1f = subset_taxa(physeq, Phylum %in%  &quot;__Firmicutes&quot; &amp; Order %in% &quot;__Selenomonadales&quot; &amp; Family %in% &quot;__Veillonellaceae&quot;)
  mphyseq = psmelt(p1f)
  mphyseq &lt;- subset(mphyseq, Abundance &gt; 0)
  ggplot(data = mphyseq, mapping = aes_string(x = &quot;Intervention&quot;, y = &quot;Abundance&quot;)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position=&quot;none&quot;)
}


plotBefore = plot_abundance(phylo_data3,
                            ylab=&quot;Abundance prior to transformation&quot;)
plotAfter = plot_abundance(phylo_data3ra,
                           ylab=&quot;Relative Abundance&quot;)
# Combine each plot into one graphic.
grid.arrange(nrow = 2, plotBefore, plotAfter)</code></pre>
<p><img src="figure/data_processing.Rmd/unnamed-chunk-2-1.png" width="225%" style="display: block; margin: auto;" /></p>
<p>Note the Genus: Allisonella &amp; Megasphaera were only present in Int. Group A.</p>
</div>
</div>
</div>
<div id="phylum-proteobacteria" class="section level3">
<h3>Phylum: Proteobacteria</h3>
<pre class="r"><code>plot_abundance = function(physeq, title = &quot;&quot;, Facet = &quot;Order&quot;, ylab=&quot;Abundance&quot;){
  # Arbitrary subset, based on Phylum, for plotting
  p1f = subset_taxa(physeq, Phylum %in%  &quot;__Proteobacteria&quot;)
  mphyseq = psmelt(p1f)
  mphyseq &lt;- subset(mphyseq, Abundance &gt; 0)
  ggplot(data = mphyseq, mapping = aes_string(x = &quot;Intervention&quot;, y = &quot;Abundance&quot;)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position=&quot;none&quot;)
}


plotBefore = plot_abundance(phylo_data3,
                            ylab=&quot;Abundance prior to transformation&quot;)
plotAfter = plot_abundance(phylo_data3ra,
                           ylab=&quot;Relative Abundance&quot;)
# Combine each plot into one graphic.
grid.arrange(nrow = 2, plotBefore, plotAfter)</code></pre>
<p><img src="figure/data_processing.Rmd/4-rel-abund-4-1.png" width="225%" style="display: block; margin: auto;" /></p>
</div>
<div id="phylum-verrucomicrobia" class="section level3">
<h3>Phylum: Verrucomicrobia</h3>
<pre class="r"><code>plot_abundance = function(physeq, title = &quot;&quot;, Facet = &quot;Family&quot;, ylab=&quot;Abundance&quot;){
  # Arbitrary subset, based on Phylum, for plotting
  p1f = subset_taxa(physeq, Phylum %in%  &quot;__Verrucomicrobia&quot;)
  mphyseq = psmelt(p1f)
  mphyseq &lt;- subset(mphyseq, Abundance &gt; 0)
  ggplot(data = mphyseq, mapping = aes_string(x = &quot;Intervention&quot;, y = &quot;Abundance&quot;)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position=&quot;none&quot;)
}


plotBefore = plot_abundance(phylo_data3,
                            ylab=&quot;Abundance prior to transformation&quot;)</code></pre>
<pre><code>Warning in prune_taxa(taxa, phy_tree(x)): prune_taxa attempted to reduce tree to 1 or fewer tips.
 tree replaced with NULL.</code></pre>
<pre class="r"><code>plotAfter = plot_abundance(phylo_data3ra,
                           ylab=&quot;Relative Abundance&quot;)</code></pre>
<pre><code>Warning in prune_taxa(taxa, phy_tree(x)): prune_taxa attempted to reduce tree to 1 or fewer tips.
 tree replaced with NULL.</code></pre>
<pre class="r"><code># Combine each plot into one graphic.
grid.arrange(nrow = 2, plotBefore, plotAfter)</code></pre>
<p><img src="figure/data_processing.Rmd/4-rel-abund-5-1.png" width="225%" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gridExtra_2.3     xtable_1.8-4      kableExtra_1.1.0  plyr_1.8.4       
 [5] data.table_1.12.6 readxl_1.3.1      forcats_0.4.0     stringr_1.4.0    
 [9] dplyr_0.8.3       purrr_0.3.3       readr_1.3.1       tidyr_1.0.0      
[13] tibble_2.1.3      ggplot2_3.2.1     tidyverse_1.3.0   lmerTest_3.1-1   
[17] lme4_1.1-21       Matrix_1.2-17     phyloseq_1.30.0  

loaded via a namespace (and not attached):
 [1] nlme_3.1-140        fs_1.3.1            lubridate_1.7.4    
 [4] webshot_0.5.2       httr_1.4.1          rprojroot_1.3-2    
 [7] numDeriv_2016.8-1.1 tools_3.6.1         backports_1.1.5    
[10] R6_2.4.1            vegan_2.5-6         DBI_1.0.0          
[13] lazyeval_0.2.2      BiocGenerics_0.32.0 mgcv_1.8-28        
[16] colorspace_1.4-1    permute_0.9-5       ade4_1.7-13        
[19] withr_2.1.2         tidyselect_0.2.5    compiler_3.6.1     
[22] git2r_0.26.1        cli_1.1.0           rvest_0.3.5        
[25] Biobase_2.46.0      xml2_1.2.2          labeling_0.3       
[28] scales_1.1.0        digest_0.6.23       minqa_1.2.4        
[31] rmarkdown_1.18      XVector_0.26.0      pkgconfig_2.0.3    
[34] htmltools_0.4.0     dbplyr_1.4.2        rlang_0.4.2        
[37] rstudioapi_0.10     farver_2.0.1        generics_0.0.2     
[40] jsonlite_1.6        magrittr_1.5        biomformat_1.14.0  
[43] Rcpp_1.0.3          munsell_0.5.0       S4Vectors_0.24.1   
[46] Rhdf5lib_1.8.0      ape_5.3             lifecycle_0.1.0    
[49] stringi_1.4.3       yaml_2.2.0          MASS_7.3-51.4      
[52] zlibbioc_1.32.0     rhdf5_2.30.1        grid_3.6.1         
[55] parallel_3.6.1      promises_1.1.0      crayon_1.3.4       
[58] lattice_0.20-38     Biostrings_2.54.0   haven_2.2.0        
[61] splines_3.6.1       multtest_2.42.0     hms_0.5.2          
[64] zeallot_0.1.0       knitr_1.26          pillar_1.4.2       
[67] igraph_1.2.4.2      boot_1.3-22         reshape2_1.4.3     
[70] codetools_0.2-16    stats4_3.6.1        reprex_0.3.0       
[73] glue_1.3.1          evaluate_0.14       modelr_0.1.5       
[76] vctrs_0.2.0         nloptr_1.2.1        httpuv_1.5.2       
[79] foreach_1.4.7       cellranger_1.1.0    gtable_0.3.0       
[82] assertthat_0.2.1    xfun_0.11           broom_0.5.2        
[85] later_1.0.0         viridisLite_0.3.0   survival_2.44-1.1  
[88] iterators_1.0.12    IRanges_2.20.1      workflowr_1.5.0    
[91] cluster_2.1.0      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
